### PR TITLE
Update renderer

### PR DESCRIPTION
**Pull request by yjchoi1**
**Describe the PR**
Update renderer as follows:
* Enable creating 3d .gif animation (new option)
* Enable creating 2d .vtk options (did not support previously)
* Enable specifying output filename when saving render result

**Related Issues/PRs**
This PR is related to [this issue](https://github.com/geoelements/gns/issues/31#issue-1448506358).

### CODE REVIEW

File: gns/render_rollout.py [hunk 1]
```diff
@@ -38,13 +9,15 @@
 import os
 from pyevtk.hl import pointsToVTK
 
-flags.DEFINE_string("rollout_path", None, help="Path to rollout pickle file.")
+flags.DEFINE_string("rollout_dir", None, help="Directory where rollout.pkl are located")
+flags.DEFINE_string("rollout_name", None, help="Name of rollout `.pkl` file")
 flags.DEFINE_integer("step_stride", 3, help="Stride of steps to skip.")
-flags.DEFINE_boolean("block_on_show", True, help="For test purposes.")
+flags.DEFINE_enum("output_mode", "gif", ["both", "gif", "vtk"], help="Type of render output")
```
Comment (kks32) [inline gns/render_rollout.py:15]: don't use `both`, can we pass `--output-mode=gif --output-mode=vtk`?

File: gns/render_rollout.py [hunk 2]
```diff
@@ -53,69 +26,193 @@
...
+        :param point_size: particle size for visualization
+        :param timestep_stride: numer of timesteps to stride for visualization (i.e., sampling rate)
+        :param vertical_camera_angle: camera angle in airplane view in 3d render
+        :param viewpoint_rotation: speed of viewpoint rotation in 3d render
+        :return: gif format animation
+        """
+        # Init figures
+        fig = plt.figure()
+        if self.dims == 2:
+            ax1 = fig.add_subplot(1, 2, 1)
```
Comment (kks32) [inline gns/render_rollout.py:98]: can we do `projection='rectilinear'`?

File: gns/render_rollout.py [hunk 3]
```diff
@@ -53,69 +26,193 @@
...
+    render = Render(input_dir=FLAGS.rollout_dir, input_name=FLAGS.rollout_name)
+
+    if FLAGS.output_mode == "both":
+        render.render_gif_animation(
+            point_size=1,
+            timestep_stride=FLAGS.step_stride,
+            vertical_camera_angle=20,
+            viewpoint_rotation=0.3
+        )
+        render.write_vtk()
```
Comment (kks32) [inline gns/render_rollout.py:204]: Remove this

**Review by kks32 — CHANGES_REQUESTED** [review]

File: gns/render_rollout.py (hunk 1 above)
Comment (yjchoi1) [reply gns/render_rollout.py]: removed `both`.

File: gns/render_rollout.py (hunk 2 above)
Comment (yjchoi1) [reply gns/render_rollout.py]: Added `rectilinear`.

File: gns/render_rollout.py [hunk 4]
```diff
@@ -53,69 +26,193 @@
...
+                                np.array(coord[:, 0]),
+                                np.array(coord[:, 1]),
+                                np.zeros_like(coord[:, 1]),
+                                data={"displacement": disp})
+                elif self.dims == 3:
+                    pointsToVTK(f"{path}/points{i}",
+                                np.array(coord[:, 0]),
+                                np.array(coord[:, 1]),
+                                np.array(coord[:, 2]),
+                                data={"displacement": disp})
```
[the remark this answers is not in the document]
Comment (yjchoi1) [reply gns/render_rollout.py]: Merged the lines.

File: gns/render_rollout.py (hunk 3 above)
Comment (yjchoi1) [reply gns/render_rollout.py]: Removed.

**Review by kks32 — APPROVED** [review]

**Pull request merged**